### PR TITLE
chore: merge chart hooks and views

### PR DESCRIPTION
**Pull request by magnew**
Closes: 

### Description:

- Merges useBarChart and usePieChart -> useSqlChart
- Merges BarChart.tsx and PieChart.tsx -> SqlRunnerChart
- Makes a parent config type for these charts

Note: doesn't give the table the same treatment. The types are more different. If we like this we can more that way. 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/backend/src/generated/routes.ts
Comment (magnew) [inline packages/backend/src/generated/routes.ts:1]: If we need this more strongly typed, I can add back a union type.

File: packages/frontend/src/features/sqlRunner/transformers/useSqlChart.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            });
+        } else {
+            throw new Error('Unknown chart type');
+        }
+    }, [config, transformer]);
+
+    return useAsync(
+        async () => visTransformer.getEchartsSpec(config),
+        [config, visTransformer],
+    );
```
Comment (ZeRego) [inline packages/frontend/src/features/sqlRunner/transformers/useSqlChart.ts:45]: These 2 hooks can be combined as they have the same dependencies.

**Review by owlas — APPROVED** [review]

**Review by ZeRego — APPROVED** [review]

File: packages/frontend/src/components/DashboardTiles/DashboardSqlChartTile.tsx [hunk 2]
```diff
@@ -91,11 +92,15 @@ export const DashboardSqlChartTile: FC<Props> = ({
...
+                    {data.chart.config.type === ChartKind.TABLE &&
+                        isTableChartSQLConfig(data.chart.config) && (
+                            <Table
+                                data={data.results}
+                                config={data.chart.config}
+                            />
+                        )}
                     {data.chart.config.type === ChartKind.VERTICAL_BAR && (
-                        <BarChart
+                        <SqlRunnerChart
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/DashboardTiles/DashboardSqlChartTile.tsx:103]: Yup, good call. I thought I had it in the saving Pie charts review, but I guess I missed it.

[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/DashboardTiles/DashboardSqlChartTile.tsx]: Done

[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/DashboardTiles/DashboardSqlChartTile.tsx]: Done

File: packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx [hunk 3]
```diff
@@ -251,7 +250,7 @@ export const ContentPanel: FC = () => {
                             )}
 
                             {queryResults?.results && pieChartConfig && (
-                                <PieChart
+                                <SqlRunnerChart
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/features/sqlRunner/components/ContentPanel.tsx:253]: Will follow up.

File: packages/frontend/src/features/sqlRunner/transformers/useSqlChart.ts (hunk 1 above)
Comment (magnew) [reply packages/frontend/src/features/sqlRunner/transformers/useSqlChart.ts:45]: You mean combine BarChartDataTransformer and PieChartDataTransformer? Can do in a follow up.

**Pull request merged**